dangerzone/document.py

```diff
@@ -157,3 +188,6 @@ def mark_as_failed(self) -> None:
     def mark_as_safe(self) -> None:
         log.debug(f"Marking doc {self.id} as 'safe'")
         self.state = Document.STATE_SAFE
+
+        if self._archive:
+            self.archive()
```

Comment:
I don't think `mark_as_safe()` should have side-effects. Like the rest of the `mark_as_*` functions, it should just update the status of the document.

I would suggest moving this to a separate method, maybe in the same place where we have the logic that copies the final document to the output directory.

Reply:
That's a much better place. Thank you. Done in https://github.com/freedomofpress/dangerzone/pull/255/commits/a8fe7afdb652c99de883574536699864580d5712

Reply:
Yay, I like it!